### PR TITLE
Added missing event results, refactored Android/IOS code and optimized to background threads

### PR DESCRIPTION
**Pull request by syndbg**
~~This is WIP.~~ Ready for review

- [x] Refactored IOS code
- [x] Added a demo project to easily try out the plugin. The existing `demo/index.html` was utilised to some extent.
- [x] Added a missing deleteEvent plugin result
- [x] Add plugin results for all other events. This is a must if you want to chain success/error callbacks with say, toast messages.
- [x] Fix some IOS code issues detected by XCode
- [x] Attempt to move the plugin to a background thread

### CODE REVIEW

File: src/ios/Calendar.h [hunk 1]
```diff
@@ -33,6 +33,6 @@
 - (void)listCalendars:(CDVInvokedUrlCommand*)command;
 
 - (void)deleteEvent:(CDVInvokedUrlCommand*)command;
-- (void)deleteEventFromNamedCalendar:(CDVInvokedUrlCommand*)command;
+- (void)deleteEventFromCalendar:(CDVInvokedUrlCommand*)command calendar: (EKCalendar *) calendar;
```
Comment (EddyVerbruggen) [inline src/ios/Calendar.h]: The JS API calls into `deleteEventFromNamedCalendar`, but this no longer matches that. Can you please elaborate?

File: src/ios/Calendar.m [hunk 2]
```diff
@@ -555,97 +569,107 @@ - (void)createEventInteractively:(CDVInvokedUrlCommand*)command {
...
+    EKEventEditViewController* controller = [[EKEventEditViewController alloc] init];
+    controller.event = myEvent;
+    controller.eventStore = self.eventStore;
+    controller.editViewDelegate = self;
+    [self.viewController presentViewController:controller animated:YES completion:nil];
+  }];
 }
 
--(void)deleteEventFromNamedCalendar:(CDVInvokedUrlCommand*)command {
+- (void) NamedCalendar:(CDVInvokedUrlCommand*)command {
```
Comment (EddyVerbruggen) [inline src/ios/Calendar.m]: This methodname seems incorrect.

**Comment by EddyVerbruggen** [comment]
Hey @syndbg great work on the changes and demo!

I've annotated your code with two comments/questions, can you check those before I merge?

Thanks,
Eddy

File: src/ios/Calendar.m (hunk 2 above)
Comment (syndbg) [inline src/ios/Calendar.m]: Pretty sure that both of these are a mess up on my part.

Comment (EddyVerbruggen) [inline src/ios/Calendar.m]: Can you fix them? I'd love to merge your PR

**Comment by syndbg** [comment]
Next, in a separate PR I will look into the Android code and do the same.

**Comment by syndbg** [comment]
Should be good now for last review and finally merge.

Major new things for the Android part are:

[x] Adding plugin results everywhere - errors and successful ones
[x] Move blocking operations in background threads
[x] Refactor the Android code. There were places with JSON parameters null checks that could easily be avoided for numbers by using optLong/Int instead of getLong/Int

**Comment by syndbg** [comment]
Are there any blockers before merging this?

**Comment by EddyVerbruggen** [comment]
Not really, I was looking for time to do so really. I'm merging it now into master and will tag a release after a few tests. Thanks again @syndbg!

File: src/ios/Calendar.m [hunk 3]
```diff
@@ -555,97 +569,107 @@ - (void)createEventInteractively:(CDVInvokedUrlCommand*)command {
...
   NSString* calendarName = [options objectForKey:@"calendarName"];
   EKCalendar* calendar = [self findEKCalendar:calendarName];
+
   if (calendar == nil) {
     CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:@"Could not find calendar"];
     [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
   } else {
     [self deleteEventFromCalendar:command calendar:calendar];
+    NSString *msg = [@"Deleted from " stringByAppendingString:calendar.title];
+    CDVPluginResult *pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_ERROR messageAsString:msg];
```
Comment (EddyVerbruggen) [inline src/ios/Calendar.m]: Woops, this is a bug, see #197

Comment (syndbg) [inline src/ios/Calendar.m]: @EddyVerbruggen Good catch! This totally slipped.

**Pull request merged**